common/djangoapps/student/views.py

```diff
@@ -1464,13 +1472,36 @@ def create_account_with_params(request, params):
         extended_profile_fields=extended_profile_fields,
         enforce_username_neq_password=True,
         enforce_password_policy=enforce_password_policy,
-        tos_required=tos_required
+        tos_required=tos_required,
     )
 
     with transaction.commit_on_success():
-        ret = _do_create_account(form)
-
-    (user, profile, registration) = ret
+        # first, create the account
+        (user, profile, registration) = _do_create_account(form)
+
+        # next, link the account with social auth, if provided
+        if should_link_with_social_auth:
+            request.user = user
```

Comment:
Instead of setting request.user here and then extracting it in ensure_user_information, user can be provided as an extra kwarg to do_auth. I think that would be cleaner.